orchagent/portsorch.cpp

```diff
@@ -345,6 +350,50 @@ bool PortsOrch::setPortMtu(sai_object_id_t id, sai_uint32_t mtu)
     return true;
 }
 
+bool PortsOrch::setHostIntfsStripTag(Port &port, sai_hostif_vlan_tag_t strip)
+{
+    SWSS_LOG_ENTER();
+    vector<Port> portv;
+
+    /*
+     * Before SAI_HOSTIF_VLAN_TAG_ORIGINAL is supported by libsai from all asic vendors,
+     * the VLAN tag on hostif is explicitly controlled with SAI_HOSTIF_VLAN_TAG_STRIP &
+     * SAI_HOSTIF_VLAN_TAG_KEEP attributes. 
+     */
+    if (port.m_type == Port::PHY)
+    {
+        portv.push_back(port);
+    }
+    else if (port.m_type == Port::LAG)
+    {
+        getLagMember(port, portv);
+    }
+    else
+    {
+        SWSS_LOG_ERROR("PortsOrch::setPortPvid port type %d not supported", port.m_type);
+        return false;
+    }
+
+    for (const auto p: portv)
+    {
+        sai_attribute_t attr;
+        attr.id = SAI_HOSTIF_ATTR_VLAN_TAG;
+        attr.value.s32 = strip;
+
+        sai_status_t status = sai_hostif_api->set_hostif_attribute(p.m_hif_id, &attr);
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            SWSS_LOG_ERROR("Failed to set %s to host interface %s",
+                        hostif_vlan_tag[strip], p.m_alias.c_str());
```

Comment:
align the spaces, or just use 8 spaces. same below.

Reply:
done

Reply:
done